### ISSUE TITLE
[TextFields] Textfield's accessibility labels or value should include placeholder, helper, error and current text

### ISSUE THREAD

**Issue by jmdetloff** [material-components/material-components-ios#4320]
Either through the label or value the placeholder helper error and current text should all be accessible.

Currently the accessibility label is never set, and the value does not include the placeholder if there is inputted text, unless there is an error in which case it seems it does. 

One potential solution:

- Placeholder + helper + error should be the label, everything else should be the value.

**Comment by jverkoey** [comment]
Most notably, the placeholder label value is not accessible and it likely should be.

**Comment by romoore** [comment]
@jverkoey Are you suggesting that the placeholder label become a distinct accessibility element or that it should be included in the accessibilityLabel?

**Comment by romoore** [comment]
Duplicate of #2822
